semattack/src/depgraph/DepGraph.cpp

```diff
@@ -253,126 +253,137 @@ std::string DepGraph::escapeLiteral(const std::string& litValue)
 }
 
 DepGraph DepGraph::parseDotFile(std::string fname) {
```

Comment:
It's better to pass c++ strings as "const std::string&" (ie a const reference to a string object) to prevent copying of the string.